### ISSUE TITLE
Failed to create cache file

### ISSUE THREAD

**Issue by nick-rackauckas** [astral-sh/ruff#9478]
Ever since upgrading to version 0.1.12, seeing this error when trying to run `ruff check`. Works when downgrading to `0.1.11`.

````
ruff failed
  Cause: Failed to create cache file '<path-to-dir>/.ruff_cache/0.1.12/16676361985305082552'
  Cause: No such file or directory (os error 2)
````

**Comment by charliermarsh** [comment]
Ah sorry, this must be me -- fixing now.

**Comment by charliermarsh** [comment]
Can you tell me any more about your project setup? Do you use a single `pyproject.toml` file at the root? Do you use `--config` to pass a `pyproject.toml`?

**Comment by nick-rackauckas** [comment]
> Can you tell me any more about your project setup? Do you use a single `pyproject.toml` file at the root? Do you use `--config` to pass a `pyproject.toml`?

A single `pyproject.toml` at the root

**Comment by charliermarsh** [comment]
I'm able to reproduce this by trying to lint a file outside of the current working directory (which also doesn't have a `pyproject.toml` file above it):

```shell
❯ ruff check ../foo.py
ruff failed
  Cause: Failed to create cache file '/Users/crmarsh/workspace/ruff/.ruff_cache/0.1.12/14778054398178272117'
  Cause: No such file or directory (os error 2)
```

**Comment by axiangcoding** [comment]
same issue. looking forward to release
